### ISSUE TITLE
 IOU - Replace receipt option is available for requestee in combined IOU report

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#39436]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.59-0
**Reproducible in staging?:** y
**Reproducible in production?:** n
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to staging.new.expensify.com
2. [User A] Go to chat with User B that has no unsettled IOU.
3. [User A] Request money from User B and attach a receipt.
4. [User B] Go to combined IOU report with User A.
5. [User B] Click on the receipt.
6. [User B] Click 3-dot menu.

## Expected Result:
There will be no Replace receipt option for User B (requestee).

## Actual Result:
Replace receipt option is available for User B (requestee).
This issue only happens in combined IOU report.

## Workaround:
n/a

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/a007796a-23c6-4792-b540-9087803e5546

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kbecciv** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Replace receipt option is available for User B (requestee).
This issue only happens in combined IOU report.

### What is the root cause of that problem?
When we open the receipt in combine report, the reportID in URL is the expense report so `canEditReceipt` check is wrong here because the parentReportAction now is the report preview action.


https://github.com/Expensify/App/blob/879378ff68dc026c1b4ba6af30b70051b99c6d35/src/components/ReportActionItem/ReportActionItemImage.tsx#L98-L101

https://github.com/Expensify/App/blob/879378ff68dc026c1b4ba6af30b70051b99c6d35/src/pages/TransactionReceiptPage.tsx#L36
### What changes do you think we should make in order to solve the problem?
We should get the transactionReportID and pass it as `reportID` param of `getRoute` function here. So all check in `TransactionReceiptPage` will work well even if we open the receipt in combine report.

https://github.com/Expensify/App/blob/879378ff68dc026c1b4ba6af30b70051b99c6d35/src/components/ReportActionItem/ReportActionItemImage.tsx#L98-L101

To do that we should store `transactionThreadReport` or `transactionThreadReportID` in `ShowContextMenuContext`.

1. Add `transactionThreadReport` in `ShowContextMenuContextProps `

```
type ShowContextMenuContextProps = {
    ...
    transactionThreadReport: OnyxEntry<Report>;
};

const ShowContextMenuContext = createContext<ShowContextMenuContextProps>({
    ...
    transactionThreadReport: null
});
```
https://github.com/Expensify/App/blob/879378ff68dc026c1b4ba6af30b70051b99c6d35/src/components/ShowContextMenuContext.ts#L12

2. Store `transactionThreadReport` into `contextValue` here.

https://github.com/Expensify/App/blob/879378ff68dc026c1b4ba6af30b70051b99c6d35/src/pages/home/report/ReportActionItem.tsx#L364

3. Here, get `transactionThreadReport` and pass `transactionThreadReport.reportID` to getRoute function 

```
{({report, transactionThreadReport}) => (
    <PressableWithoutFocus
        style={[styles.w100, styles.h100, styles.noOutline as ViewStyle]}
        onPress={() => Navigation.navigate(ROUTES.TRANSACTION_RECEIPT.getRoute(transactionThreadReport?.reportID ?? report?.reportID ?? '', transaction?.transactionID ?? ''))}
        accessibilityLabel={translate('accessibilityHints.viewAttachment')}
        accessibilityRole={CONST.ROLE.BUTTON}
    >
        <ReceiptImage {...propsObj} />
    </PressableWithoutFocus>
)}
```
https://github.com/Expensify/App/blob/879378ff68dc026c1b4ba6af30b70051b99c6d35/src/components/ReportActionItem/ReportActionItemImage.tsx#L98-L101

### What alternative solutions did you explore? (Optional)
NA

**Comment by jasperhuangg** [comment]
I don't think this is worth blocking deploy on this as we're marking other bugs related to the combined IOU report view as non-blockers.

**Comment by nkdengineer** [comment]
@jasperhuangg The PR is [here](https://github.com/Expensify/App/pull/39572)

**Comment by abdulrahuman5196** [comment]
@jasperhuangg Wanted to check if C+ is needed here? Seems I was removed and deployment blocker is also removed?

**Comment by alexpensify** [comment]
@nkdengineer - I've sent an offer via Upwork. Please accept and I can complete the payment process. Thanks!

**Comment by alexpensify** [comment]
Closing, I was able to complete the payment to @nkdengineer. @allroundexperts please submit a request in Chat. Thanks!

**Comment by JmillsExpensify** [comment]
$500 approved for @allroundexperts
